### PR TITLE
Clear SetCell write tracker matrix in shared matrix which is outside of collab window

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Clear SetCell write tracker matrix in shared matrix which is outside of collab window.
Follow up on the implementation of FWW policy for the shared matrix. As part of that implementation, we create a tracker matrix to store the seq number and client id at which a change in the cell is made. However we only need a part, of the matrix which is only in collab window and we can clear the rest.

### CODE REVIEW

File: packages/dds/matrix/src/matrix.ts [hunk 1]
```diff
@@ -529,6 +532,36 @@ export class SharedMatrix<T = any>
 
 		// Only need to store it in the snapshot if we have switched the policy already.
 		if (this.setCellLwwToFwwPolicySwitchOpSeqNumber > -1) {
+			const cellLastWriteTrackerToSummarize = new SparseArray2D<CellLastWriteTrackerItem>();
+			let count = 100000;
```
Comment (vladsud) [inline packages/dds/matrix/src/matrix.ts:536]: Not sure what it is for/ I see it only decremented, but never read from

Comment (vladsud) [reply packages/dds/matrix/src/matrix.ts:536]: I think you likely wanted to say - we will scan that many rows?
Might be better to be count of cells, not rows, as table theoretically could have a lot of columns

File: packages/dds/matrix/src/matrix.ts [hunk 2]
```diff
@@ -529,6 +532,36 @@ export class SharedMatrix<T = any>
...
+				const rowHandle = this.rowHandles.getHandle(r);
+				if (isHandleValid(rowHandle)) {
+					for (let c = 0; c < this.colCount; c++) {
+						count--;
+						const colHandle = this.colHandles.getHandle(c);
+						if (isHandleValid(colHandle)) {
+							const item = this.cellLastWriteTracker.getCell(rowHandle, colHandle);
+							if (
+								item !== undefined &&
+								(item.seqNum > this.rows.getCollabWindow().minSeq ||
```
Comment (vladsud) [inline packages/dds/matrix/src/matrix.ts:552]: I'd highly recommend you to cache the value of "this.rows.getCollabWindow().minSeq" and not call it twice on each iteration.

Comment (vladsud) [reply packages/dds/matrix/src/matrix.ts:552]: I wonder why this.cols.getCollabWindow().minSeq and this.rows.getCollabWindow().minSeq could be different. But it does not matter - we can use max() of them

File: packages/dds/matrix/src/matrix.ts (hunk 1 above)
Comment (vladsud) [reply packages/dds/matrix/src/matrix.ts:536]: That said, given that you do not change it in place, but rather create a copy (cellLastWriteTrackerToSummarize), you can't bail out sooner, right?

Comment (jatgarg) [reply packages/dds/matrix/src/matrix.ts:536]: yeah, there is not in place freeing up the cells. I don't think we can do it partially and also cannot do it completely for bigger matrices. Closing this PR for now.

**Pull request closed without merging**